utils/generic-components.stanza

```diff
@@ -1194,15 +1211,22 @@ public pcb-module inverted-f-antenna :
 ; Adds an RGB led to a processor using PWM pins
 public defn add-rgb-indicator (brightness:Double, proc:JITXObject, vin:JITXObject):
   inside pcb-module:
-    val vio = 3.3 ; TODO: get this from the vin pin property
-    ; Calculate resistance to get appropriate brightness
-    defn calc-r (l:JITXObject, brightness:Double, vio:Double):
-      val i = PWL(property(l.mcd-current))[brightness]
-      closest-std-val((vio - property(l.forward-voltage)) / i, 1.0)
-    
-    public inst led : ocdb/foshan-optoelectronics/FM-B2020RGBA-HG/component ; TODO: make generic
+    pcb-module rgb-indicator :
+      port pwm : timer[3]
+      val vio = 3.3 ; TODO: get this from the vin pin property
+      ; Calculate resistance to get appropriate brightness
+      defn calc-r (l:JITXObject, brightness:Double, vio:Double):
+        val i = PWL(property(l.mcd-current))[brightness]
+        closest-std-val((vio - property(l.forward-voltage)) / i, 5.0)
+      public inst led : ocdb/foshan-optoelectronics/FM-B2020RGBA-HG/component ; TODO: make generic
+      
+      res-strap(pwm[0].timer, led.l.r, calc-r(led.l.r, brightness, vio))
+      res-strap(pwm[1].timer, led.l.g, calc-r(led.l.g, brightness, vio))
+      res-strap(pwm[2].timer, led.l.b, calc-r(led.l.b, brightness, vio))
+
+    inst rgb:rgb-indicator
     require pwm:timer[3] from proc
```

Comment:
You have a port pwm : timer[3] and then you overload pwm with a require statement? I am confused about which one gets connected to the rgb.

Reply:
Good point - I've changed the names to avoid conflating timer and pwm

A module is defined with a port `control-pins` of type `timer[3]`. Later I use a require statement to pull a port of that type of the processor and connect it to the module port. So in the end, three pins supporting timer are attached to the led.